### ISSUE TITLE
[no-unused-vars] global never used

### ISSUE THREAD

**Issue by Tansito** [typescript-eslint/typescript-eslint#2558]
**Repro**

Updating `@typescript-eslint/eslint-plugin` and `@typescript-eslint/parser` from **3.10.0** to **4.1.0** the linter is showing the next message:

```
warning  'global' is defined but never used  @typescript-eslint/no-unused-vars
```

Our code is the next:
```TS
import { Tab } from './Tab';

customElements.define('custom-tab', Tab);

declare global {
  interface HTMLElementTagNameMap {
    'custom-tab': Tab;
  }
}
```

And we solved the problem thanks to the next rule:

```JSON
"rules": {
  "@typescript-eslint/no-unused-vars": [
    "error",
    {
      "varsIgnorePattern": "^global$"
    }
  ]
}
```

Our current rules for the project: https://github.com/IBMResearch/eslint-config-ibmresearch

**Expected Result**

How in this case the `global` reserved word references to the _global_ scope, this message should not be shown as a warning.

**Actual Result**

The plugin is showing `global` as an unused variable.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.1.0` |
| `@typescript-eslint/parser`        | `4.1.0` |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | `7.9.0` |
| `node`                             | `14.10.1` |

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/issues/2523 et al.

Please use the issue search.
